ext/MooncakeFluxExt.jl

```diff
@@ -0,0 +1,28 @@
+module MooncakeFluxExt
+
+using Mooncake, Flux
+using Base: IEEEFloat
+import Mooncake: DefaultCtx, rrule!!, @is_primitive, CoDual, zero_fcodual, NoRData
+
+@is_primitive DefaultCtx Tuple{
+    typeof(Flux.Losses.mse),Array{P},Array{P}
+} where {P<:IEEEFloat}
+
+function rrule!!(
+    ::CoDual{typeof(Flux.Losses.mse)}, X::CoDual{<:Array{P}}, Y::CoDual{<:Array{P}}
+) where {P<:IEEEFloat}
+    norm_factor = P(2) / P(length(X.x)) .* (X.x .- Y.x)
+
+    function flux_mse_pullback(dloss::P)
+        # adjoints got by VJP reverse pass equations.
+        temp = norm_factor .* dloss
+        @. X.dx += temp
+        @. Y.dx -= temp
```

Comment:
```suggestion
        tmp = dloss * P(2 / length(X.x))
        @inbounds for n in eachindex(X.x)
            d = X.x[n] - Y.x[n]
            X.dx[n] += tmp * d
            Y.dx[n] -= tmp * d
        end
```
I did some benchmarking locally, and I think this might be the best way to implement the reverse-pass. It avoids creating a new `Array{P}`, and still only loads the memory from each of the arrays involved into memory a single time.

Could you let me know whether you are also seeing that this gives better performance than the current implementation?

Reply:
yup you are correct. ive changed it now, thanks!